module/zfs/zpl_inode.c

```diff
@@ -410,13 +410,14 @@ zpl_rename2(struct inode *sdip, struct dentry *sdentry,
 	int error;
 	fstrans_cookie_t cookie;
 
-	/* We don't have renameat2(2) support */
-	if (flags)
+	/* We only support flags which are needed for overlayfs-on-ZFS. */
```

Comment:
Are there currently other flags that you're not supporting?

Reply:
No, but the issue is that `renameat(2)` defers to the filesystem to support `RENAME_*` flags (since only the filesystem can do it atomically). So in a future kernel version they might add additional flags which we might not support, and we don't want to silently ignore them.

It's a bit of a failure on the part of Linux's VFS (filesystems don't specify what flags they support) but we're stuck with it for the moment.

Reply:
In that case, I'd suggest dropping (or rewording) the comment. Checking the flags are only those we support is a common idiom and perfectly reasonable.